File: LV 2/124 나라의 숫자.py
def convert_four(n):
    s = ''
    idx = 0
    while n != 0: 
        remainder = n % 3
        q = n // 3
        if n % 3 == 0:
            remainder = 4
            q = q - 1
        s = str(remainder) + s
        n = q

    return s

def solution(n):
    answer = ''
    k = convert_four(n)
    answer = k
    return answer

File: LV 2/124 나라의 숫자.java
class Solution {
    public String makeDigit(int n) {
        StringBuilder result = new StringBuilder();
        int carry = 0;
        while (n > 0) {
            if (n % 3 == 0) {
                result.insert(0, 4);
                n = n / 3;
                n--;
                continue;
            }
            result.insert(0, (n % 3));
            n = n / 3;
        }
        return result.toString();
    }

    public String solution(int n) {
        String answer = makeDigit(n);
        return answer;
    }
}
